python/lsst/afw/table/_syntax.py

```diff
@@ -202,3 +202,70 @@ def processArray(a):
         else:
             d[name] = processArray(self.get(schemaItem.key))
     return d
+
+def BaseCatalog_asAstropy(self, cls=None, copy=False, unviewable="copy"):
```

Comment:
I'm fairly ambivalent about underscores vs. CamelCase here; overall I have a slight preference for switching everything to underscores, but I don't want to switch just some interfaces because then it becomes hard to remember which convention to use where.  But I also don't want people to be stuck trying to remember whether it's `asAstropy` or `asAstroPy`.  Maybe let's see if we get any more opinions?

In any case, I'd like to avoid appending "table"; I think that's pretty strongly implied by the fact that this is a Catalog object, and I do think this is something people may end up typing a fair amount in interactive use, so fewer characters matters.